src/main.rs

```diff
@@ -147,35 +230,126 @@ fn main() {
                     MouseMode::CameraLook => renderer.camera.move_on_orbit(mouse_delta),
                     MouseMode::CameraPan => renderer.camera.move_focus(mouse_delta),
                 }
+                ui_state.camera_target[0] = renderer.camera.target[0];
+                ui_state.camera_target[1] = renderer.camera.target[1];
+                ui_state.camera_target[2] = renderer.camera.target[2];
```

Comment:
Can you make both of these the same type you can just assign them?
Or can you write a conversion function somewhere instead of having to do this piecemeal?